### ISSUE TITLE
Preview feature feedback: Prisma count by filtered relation (`filteredRelationCount`)

### ISSUE THREAD

**Issue by matthewmueller** [prisma/prisma#15069]
Please share your feedback about the `filteredRelationCount ` functionality released in [v4.3.0](https://github.com/prisma/prisma/releases/tag/4.3.0) in this issue.

- If you encounter a bug, please [open a bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) in this repo.
- If the feature is working well for you, please share this in a comment below or leave a 👍 on this issue.

If you have any questions, don't hesitate to ask them in the [`#prisma-client`](https://prisma.slack.com/archives/CCWDULGUW) channel in the [Prisma Slack](https://slack.prisma.io).

**Comment by chason** [comment]
This works great except that without the ability to have multiple `_count` annotations for the same relation[^1] the use cases are limited

[^1]: Captured in this feature request https://github.com/prisma/prisma/issues/15423

**Comment by sneko** [comment]
@matthewmueller did you think about allowing multiple counts for the same relation? I agree with @chason it's limited otherwise.

I guess it would make the typing a bit more complicated since multiple counts probably imply multiple names?

**Comment by ssukienn** [comment]
Good feature, great job!

For cases where I am interested in `_count` as an extra additional column, the `select` API placement makes it less ergonomic as I need to manually specify all of the remaining columns interesting for me from the model (and nested models). This is probably more related to the `select` semantics but count here is in the scope of the `findMany` operation, so opting out from all columns when specifying an "extra" `_count` creates a struggle (think about a table with 20 cols which you need to retype in select block).

**Comment by janpio** [comment]
@michalbundyra Thanks for the mention, @ruheni fixed that in our old release notes.

**Comment by SC0d3r** [comment]
Is there a way that this _count returns the total number of users not just its relations? I need it for pagination, currently I have to send 2 requests: prisma.user.findMany({,,.}) to get the users, prisma.user.count({...}) to get the total number of users

```js
await prisma.user.findMany({
  select: {
    _count: {
      select: {
        posts: { where: { title: 'Hello!' } },
      },
    },
  },
})
```

**Comment by inferusvv** [comment]
There are some differences in `not`.

In common `not`:

```js
where: {
  something: {
    not: null
  }
}
```

But with filteredRelationCount `isNot`:

```js
where: {
  something: {
    isNot: null
  }
}
```

**Comment by Weakky** [comment]
Hey @inferusvv,

Would you mind sharing a Prisma Schema that reproduces your issue? I'm mostly interested in the type of `something`. Thank you 🙏🏻

**Comment by inferusvv** [comment]
@Weakky Sorry, I am not using this flag for now, but just implement basic count with this feature enabled. There is just no consistency in naming.

**Comment by Weakky** [comment]
I'm afraid I'm unable to reproduce your issue.

```ts
prisma.user.findMany({
  where: { address: { isNot: { user: { isNot: { id: 1 } } } } },
  includes: {
    _count: { where: { user: { isNot: { address: { isNot: { id: 1 } } } } } }
  }
})
```

We're aware the naming is inconsistent between scalar and relations, but that has no relevance with `filteredRelationCount`.

**Comment by inferusvv** [comment]
Got. This is weired. Here is my model:

```
model PointOfSale {
  id                       String   @id @default(auto()) @map("_id") @db.ObjectId

  menu           Menu?         @relation(fields: [menuId], references: [id])
  menuId         String?       @db.ObjectId
}
```

And prisma request:

```
      this.prismaService.pointOfSale.findMany({
        where: {
          menuId: {
            not: null,
          },
        },
      });
```

There is no `isNot`, only `not`

**Comment by Weakky** [comment]
Thanks for the follow-up! That confirms my thoughts. As mentioned above, scalar fields and relations indeed have a different `not`, but that inconsistency should be equally present when `filteredRelationCount` is not enabled.

To-one relations use `isNot` to stay consistent with the `is`. I'm honestly not entirely sure why we introduced `is` for to-one relations in the first place. 

To conclude, this is a completely different topic, which won't be treated in the scope of making this feature GA.

**Comment by joshuaavalon** [comment]
Will there be a way to order by `_count`?

For example, I want to sort the post by the number of matching tags.

```ts
db.post.findMany({
  include: {
    _count: {
      select: {
        tags: {
          where: { name: { in: tags } }
        }
      }
    },
    tags: true
  },
  orderBy: [
    { tags: { _count: "desc" } } // <- this only sort by all tag count
  ]
})
```

**Comment by janpio** [comment]
Thanks everyone, `filteredRedlationCount` is Generally Available since [4.16.0](https://github.com/prisma/prisma/releases/4.16.0) 🥳 
If you have further feedback, you can open a new issue now. Thanks again!
